packages/react-reconciler/src/ReactFiberCompleteWork.old.js

```diff
@@ -1499,9 +1500,8 @@ function completeWork(
         const prevIsHidden = prevState !== null;
         if (
           prevIsHidden !== nextIsHidden &&
-          newProps.mode !== 'unstable-defer-without-hiding' &&
```

Comment:
So this checks for `unstable-defer-without-hiding` on a regular Offscreen component because this doesn't kick in for LegacyHiddenComponent. My understanding is that this option should not be available on regular Offscreen components even at FB. So we can just exclude this check.